### PR TITLE
A0-1546: Include benchmarks in unit test suite

### PR DESCRIPTION
**Pull request by pmikolajczyk41**
Pretty self-explanatory

### CODE REVIEW

**Comment by deuszx** [comment]
Aren't benchmarks a long running process? If so then maybe having it ran as part of unit test suite (on every commit) is not the best idea. How about we run it just before the merge (on merge commit)? We don't act on the results of benchmarks and having it run anyway on every PR will give us the traceability we desire.

**Comment by pmikolajczyk41** [comment]
@deuszx benchmarking is extensive due to repeating; the cases themselves go very fast:
```
running 27 tests
test tests::setup::__construct_runtime_integrity_test::runtime_integrity_tests ... ok
test tests::suite::stores_vk_with_fresh_identifier ... ok
test tests::suite::does_not_overwrite_registered_key ... ok
test tests::suite::does_not_store_too_long_key ... ok
test tests::suite::verify_shouts_when_no_key_was_registered ... ok
test tests::suite::verify_shouts_when_key_is_not_decodable ... ok
test benchmarking::suite::bench_store_key ... ok
test tests::suite::verify_shouts_when_proof_is_not_decodable ... ok
test tests::suite::verify_shouts_when_input_is_not_decodable ... ok
test benchmarking::suite::bench_verify_groth16_xor ... ok
test benchmarking::suite::bench_verify_groth16_linear_equation ... ok
test benchmarking::suite::bench_verify_groth16_merkle_tree_1024 ... ok
test benchmarking::suite::bench_verify_gm17_linear_equation ... ok
test tests::suite::verifies_proof ... ok
test benchmarking::suite::bench_verify_groth16_merkle_tree_64 ... ok
test benchmarking::suite::bench_verify_groth16_merkle_tree_8 ... ok
test tests::suite::verify_shouts_when_proof_is_incorrect ... ok
test benchmarking::suite::bench_verify_gm17_merkle_tree_1024 ... ok
test benchmarking::suite::bench_verify_gm17_merkle_tree_8 ... ok
test tests::suite::verify_shouts_when_verification_fails ... ok
test benchmarking::suite::bench_verify_gm17_xor ... ok
test benchmarking::suite::bench_verify_gm17_merkle_tree_64 ... ok
test benchmarking::suite::bench_verify_marlin_linear_equation ... ok
test benchmarking::suite::bench_verify_marlin_xor ... ok
test benchmarking::suite::bench_verify_marlin_merkle_tree_64 ... ok
test benchmarking::suite::bench_verify_marlin_merkle_tree_8 ... ok
test benchmarking::suite::bench_verify_marlin_merkle_tree_1024 ... ok

test result: ok. 27 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.04s
```

**Comment by pmikolajczyk41** [comment]
or actually 1.46s on lame GH runner: https://github.com/Cardinal-Cryptography/aleph-node/actions/runs/3463676349/jobs/5784243035#step:9:811

File: pallets/snarcos/src/benchmarking/suite.rs [hunk 1]
```diff
@@ -103,4 +106,7 @@ benchmarks! {
         let _ = insert_key::<T>(key);
     } : verify(caller::<T>(), IDENTIFIER, proof, input, Marlin)
 
+    // Benchmarks as unit tests
+
```
Comment (mike1729) [inline pallets/snarcos/src/benchmarking/suite.rs:110]: nit: unnecessary blank link

**Review by mike1729 — APPROVED** [review]

File: pallets/snarcos/src/benchmarking/suite.rs (hunk 1 above)
Comment (pmikolajczyk41) [reply pallets/snarcos/src/benchmarking/suite.rs:110]: yeah, but it is consistent with other such 'section-markers' within this long suite so I think I will left it here

**Pull request merged**